testbench/common.py

```diff
@@ -644,27 +647,30 @@ def _extract_data(data):
     return data
 
 
-def __get_streamer_response_fn(database, method, conn, test_id):
+def __get_streamer_response_fn(database, method, conn, test_id, limit=4, chunk_size=4):
     def response_handler(data):
         def streamer():
-            database.dequeue_next_instruction(test_id, method)
             d = _extract_data(data)
-            chunk_size = 4
+            bytes_yield = 0
             for r in range(0, len(d), chunk_size):
-                if r < 10:
+                if bytes_yield < limit:
                     chunk_end = min(r + chunk_size, len(d))
+                    chunk_downloaded = chunk_end - r
+                    bytes_yield += chunk_downloaded
                     yield d[r:chunk_end]
-                if conn is not None:
-                    conn.close()
-                # This exception is raised to abort the flow control. The
-                # connection has already been closed, causing the client to
-                # receive a "connection reset by peer" (or a similar error).
-                # The exception is useful in unit tests (where there is no
-                # socket to close), and stops the testbench from trying to
-                # complete a request that we intentionally aborted.
-                raise testbench.error.RestException(
-                    "Injected 'connection reset by peer' fault", 500
-                )
+                if bytes_yield >= limit:
```

Comment:
This is just "else" I guess?

Reply:
hmmmm I guess so... does the server send back anything else after the payload? 

Is it possible to `return-broken-stream-after-256K` when downloading an exact 256K object with `chunk_size=256*1024` (the download is done in one chunk and then broken stream error)? I wrote two separate conditional checks intentionally for a test case like this.